collect_app/src/androidTest/java/org/odk/collect/android/support/rules/TestRuleChain.kt

```diff
@@ -37,4 +29,26 @@ object TestRuleChain {
                 )
             )
     }
+
+    private fun createGrantPermissionRule() =
```

Comment:
Wouldn't this be more readable:

```java
private fun createGrantPermissionRule() {
        GrantPermissionRule.grant(
            Manifest.permission.ACCESS_FINE_LOCATION,
            Manifest.permission.ACCESS_COARSE_LOCATION,
            Manifest.permission.READ_EXTERNAL_STORAGE,
            Manifest.permission.CAMERA,
            Manifest.permission.RECORD_AUDIO,
            Manifest.permission.GET_ACCOUNTS
        )
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            GrantPermissionRule.grant(Manifest.permission.POST_NOTIFICATIONS)
        }
    }
```

Reply:
`GrantPermissionRule.grant` is a factory method that returns a `GrantPermissionRule` (that will grant the passed permissions), so this wouldn't work. There annoyingly isn't a "builder" approach that would make this simpler: an instance `GrantPermissionRule` is immutable.

Reply:
Unfortunately yes. Requesting `POST_NOTIFICATIONS` on older APIs throws an exception.